### PR TITLE
fix(upgarde): scale down jiva contoller deployment only if not upgraded

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does / why we need it**:
This PR fixes the issue which might occur if the upgrade job is run after successful completion. It might scale down the controller which will not be patched and remained scaled-down forever.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/upgrade/upgrader/jiva_upgrade.go [hunk 1]
```diff
@@ -517,17 +517,20 @@ func (j *jivaVolumeOptions) replicaUpgrade(pvName, openebsNamespace string) erro
...
-	// Scaling down controller ensures no I/O occurs
-	// which make volume to come in RW mode early
-	err = scaleDeploy(j.controllerObj.name, j.ns, ctrlDeployLabel, 0)
-	if err != nil {
-		statusObj.Message = "failed to get scale down target deployment"
-		statusObj.Reason = strings.Replace(err.Error(), ":", "", -1)
-		j.utaskObj, uerr = updateUpgradeDetailedStatus(j.utaskObj, statusObj, openebsNamespace)
-		if uerr != nil && isENVPresent {
-			return uerr
+	// Scale down controller deployment only if the controller is not upgraded.
```
Comment (vishnuitta) [inline pkg/upgrade/upgrader/jiva_upgrade.go:520]: should we consider returning from this function if controller's version is upgradeVersion?

Comment (shubham14bajpai) [reply pkg/upgrade/upgrader/jiva_upgrade.go:520]: In that case, success status will not be patched for REPLICA_UPGRADE step in the upgradeTask CR and will be stuck in pending state.

Comment (vishnuitta) [reply pkg/upgrade/upgrader/jiva_upgrade.go:520]: ok.. we can patch the upgradeTask CR also at the end

Comment (shubham14bajpai) [reply pkg/upgrade/upgrader/jiva_upgrade.go]: done

**Pull request merged**